src/Extensions/TimedScopes/Internal/TimedScopeEventSender.cs

```diff
@@ -20,31 +23,45 @@ public TimedScopeEventSender(TimedScopeEventSource eventSource, IHostEnvironment
 		}
 
 
-		public void LogTimedScopeEndEvent(TimedScope scope)
+		public void LogActivityStop(Activity activity)
 		{
 			if (!m_eventSource.IsEnabled())
 			{
 				return;
 			}
 
 			string serviceName = m_serviceName;
-			string subtype = scope.SubType;
-			string metadata = scope.Metadata;
-			TimedScopeResult result = scope.Result;
-			Activity activity = scope.Activity;
 			string name = activity.OperationName;
 			string correlationId = activity.Id;
 			double durationMs = activity.Duration.TotalMilliseconds;
 			string userHash = activity.GetUserHash(); //TODO: We need add middleware that will set userhash in compliant way and IsTransaction GitHub Issue #166
 			bool isTransaction = activity.IsTransaction();
 
+			string subtype = NullPlaceholder;
+			string metadata = NullPlaceholder;
+			string resultAsString = NullPlaceholder;
+			foreach (KeyValuePair<string, string> pair in activity.Tags)
+			{
+				if (string.Equals(ActivityTagKeys.Result, pair.Key, StringComparison.Ordinal))
+				{
+					resultAsString = pair.Value;
+				}
+				else if (string.Equals(ActivityTagKeys.SubType, pair.Key, StringComparison.Ordinal))
+				{
+					subtype = pair.Value;
+				}
+				else if (string.Equals(ActivityTagKeys.Metadata, pair.Key, StringComparison.Ordinal))
+				{
+					metadata = pair.Value;
+				}
```

Comment:
A `switch` statement would be nicer here, particularly since it's case insensitive. Just be warned that it will be less tolerant of `null`s

Reply:
I have not managed to make switch work here because `ActivityTagKeys.Metadata` and other properties are read-only and not a constant, and it's a bit risky to make them constants since they are used in different assemblies.